core/lib/mcxtx.h

```diff
@@ -275,4 +275,6 @@
          */
         dr_zmm_t ymm[MCXT_NUM_SIMD_SLOTS];
     };
+    /** Storage for #MCXT_NUM_OPMASK_SLOTS mask registers as part of AVX-512. */
+    dr_opmask_t opmask[MCXT_NUM_OPMASK_SLOTS];
```

Comment:
(The offset will change in the next iteration that bumps the simd count or size, but it can't be helped, and the struct size will let us have compat code.)

Reply:
Yes, but I think this is far away. AMD is not even at AVX-512 and I think their latest Ryzen just moved from double-pumped AVX (2x128) to real AVX internally. I would expect rather than having a Xeon with wider registers maybe more specialized HW coming up, like AI co-processors and such?! But know knows?! That was just a statement right, nothing to do here?

Reply:
Considering how long this code base has been around, it is worth planning for.  But we already have the solution in place: the struct size field.